### PR TITLE
Svelte 3 support

### PR DESCRIPTION
**Pull request by DeMoorJasper**
Based of the work by @antony #35

### CODE REVIEW

File: src/svelte-asset.js [hunk 1]
```diff
@@ -34,40 +40,46 @@ class SvelteAsset extends Asset {
...
+    let config = (await super.getConfig(['.svelterc', 'svelte.config.js', 'package.json'])) || {};
+    config = config.svelte || config;
+
+    let defaultOptions = {
       generate: 'dom',
-      format: 'cjs',
       store: true,
-      css: true
+      css: true,
+      format: 'cjs'
```
Comment (DeMoorJasper) [inline src/svelte-asset.js]: Defaulting this to es || esm didn't work :(

File: test/basic.js [hunk 2]
```diff
@@ -17,4 +17,10 @@ describe('basic', function() {
       ]
     });
   });
+
+  // Not sure if we can will be able to run this test
+  it.skip('Should support Svelte 3 features', async function() {
```
Comment (DeMoorJasper) [inline test/basic.js]: We do need to figure out how to test this, perhaps write a util to change installed svelte version or something, not sure how this is gonna work out

Comment (antony) [reply test/basic.js]: I would suggest a couple of node modules with their own `package.json` inside the test dir maybe. worth an experiment!

File: src/svelte-asset.js (hunk 1 above)
Comment (antony) [reply src/svelte-asset.js]: Strange. The Svelte 3 compiler only recognises `esm`, `es` is v2. Maybe it's the same issue as discussed with dependency versions

File: test/basic.js (hunk 2 above)
Comment (DeMoorJasper) [reply test/basic.js]: Sounds like a good idea, I'll experiment a lil with that idea. If it fails I guess we'll have to port all tests to Svelte 3 and drop tests for Svelte 2

Comment (DeMoorJasper) [reply test/basic.js]: Found some hacky way to implement this using a monorepo and some weird commands

File: src/svelte-asset.js (hunk 1 above)
Comment (DeMoorJasper) [reply src/svelte-asset.js]: Not entirely sure why it happens but it has something to do with the HMR code, probably the wrapper that breaks imports

Comment (DeMoorJasper) [reply src/svelte-asset.js]: I think I figured it out, the regex in makeHot was only working for commonjs

**Comment by DeMoorJasper** [comment]
Everything should work for V2 and V3 now :)

Would be great if anyone would be willing to stress test these changes a lil before I merge these into master and publish a release.

**Comment by antony** [comment]
@DeMoorJasper I'll test these as soon as I am able to - thanks for your hard work!

**Pull request merged**